tests/bot/bot.spec.js

```diff
@@ -0,0 +1,180 @@
+      amount: 111,
+      fee: 0.111,
+      creator_id: '61006f0e85ad4f96cde94141',
+      seller_id: '61006f0e85ad4f96cde94141',
+      type: 'sell',
+      status: 'PENDING',
+      fiat_amount: 111,
+      fiat_code: 'ves',
+      payment_method: 'Pagomovil',
+      tg_chat_id: '1',
+      tg_order_message: '1',
+      created_at: '2021-08-30T20:52:33.870Z',
+    };
+    sandbox = sinon.createSandbox();
+    // Mock process.env
+    sandbox.stub(process, 'env').value({
+      CHANNEL: '@testChannel',
+      MIN_PAYMENT_AMT: 100,
+      NODE_ENV: 'test',
+      INVOICE_EXPIRATION_WINDOW: 3600000,
+      LND_GRPC_HOST: '127.0.0.1:10005',
+    });
+
+    // Mock TelegramServer
+    server = {
+      getClient: sandbox.stub().returns({
+        makeCommand: sandbox.stub().returns({}),
+        sendCommand: sandbox.stub().resolves({ ok: true }),
+        getUpdates: sandbox.stub().resolves({
+          ok: true,
+          result: [{
+            update_id: 1,
+            message: {
+              message_id: 1,
+              from: {
+                id: 1,
+                is_bot: false,
+                first_name: 'Test',
+                username: 'testuser',
+                language_code: 'en',
+              },
+              chat: {
+                id: 1,
+                first_name: 'Test',
+                username: 'testuser',
+                type: 'private',
+              },
+              date: 1678888888,
+              text: '/start',
+              entities: [{
+                offset: 0,
+                length: 6,
+                type: 'bot_command',
+              }],
+            },
+          }],
+        }),
+        sendMessage: sandbox.stub().resolves({ ok: true }),
+      }),
+      ApiURL: 'http://localhost:9001',
+    };
+
+    // Mock mongo connection and models
+    sandbox.stub(User, 'findOne').resolves(user);
+    sandbox.stub(Order, 'findOne').resolves(null);
+    sandbox.stub(Order.prototype, 'save').resolves(order);
+
+    // Initialize the bot
+    initialize(token, { telegram: { apiRoot: server.ApiURL } });
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('should start', async () => {
+    const client = server.getClient(token, { timeout: 5000 });
+    const command = client.makeCommand('/start');
+    const res = await client.sendCommand(command);
+    expect(res.ok).to.be.equal(true);
+    const updates = await client.getUpdates();
+    expect(updates.ok).to.be.equal(true);
+    expect(updates.result.length).to.be.equal(1);
+  });
+
+  it('should return /sell help', async () => {
+    const client = server.getClient(token, { timeout: 5000 });
+    const command = client.makeCommand('/sell help');
+    const res = await client.sendCommand(command);
+    expect(res.ok).to.be.equal(true);
+  });
+
+  it('should create a /sell', async () => {
+    const client = server.getClient(token, { timeout: 5000 });
+    const command = client.makeCommand('/sell 100 1 ves Pagomovil');
```

Comment:
I have a doubt about this test, if I change the commando to a wrong command, let's say `client.makeCommand('/sello 100 1 ves Pagomovil');` the test still passes, what are we testing here? can we make the test fails if the command is wrong?

Reply:
I will add the fail tests, the reason is that the telegram api is mocked with a certain response, for the fails tests it's necessary to add the mocks for the failed tests

Reply:
The unit tests have a controlled environment, we need to add all the mocked data to makes the test run every possible scenario

Reply:
Yes, I'm working on that

Reply:
this is an amazing job, I tested and I got this on validateInvoice
```log
    validateInvoice
[2024-10-17T15:24:24.364-03:00] error: ExpectedLnPrefix Error: ExpectedLnPrefix
    at module.exports (/home/grunch/dev/bot-danfer/node_modules/invoices/bolt11/parse_payment_request.js:68:11)
    at /home/grunch/dev/bot-danfer/bot/validations.js:453:27
    at step (/home/grunch/dev/bot-danfer/bot/validations.js:56:23)
    at Object.next (/home/grunch/dev/bot-danfer/bot/validations.js:37:53)
    at /home/grunch/dev/bot-danfer/bot/validations.js:31:71
    at new Promise (<anonymous>)
    at __awaiter (/home/grunch/dev/bot-danfer/bot/validations.js:27:12)
    at validateInvoice (/home/grunch/dev/bot-danfer/bot/validations.js:446:58)
    at Context.<anonymous> (/home/grunch/dev/bot-danfer/tests/bot/validation.spec.js:510:25)
    at callFn (/home/grunch/dev/bot-danfer/node_modules/mocha/lib/runnable.js:366:21)
    at Runnable.run (/home/grunch/dev/bot-danfer/node_modules/mocha/lib/runnable.js:354:5)
    at Runner.runTest (/home/grunch/dev/bot-danfer/node_modules/mocha/lib/runner.js:678:10)
    at /home/grunch/dev/bot-danfer/node_modules/mocha/lib/runner.js:801:12
    at next (/home/grunch/dev/bot-danfer/node_modules/mocha/lib/runner.js:593:14)
    at /home/grunch/dev/bot-danfer/node_modules/mocha/lib/runner.js:603:7
    at next (/home/grunch/dev/bot-danfer/node_modules/mocha/lib/runner.js:486:14)
    at Immediate._onImmediate (/home/grunch/dev/bot-danfer/node_modules/mocha/lib/runner.js:571:5)
    at process.processImmediate (node:internal/timers:478:21)
[2024-10-17T15:24:24.365-03:00] debug: invalid-invoice 
```
After fixing this I think we can merge it 😃

Reply:
This is an error that is part of the implementation, the error is raised in the method parsePaymentRequest and it goes to the catch, that is why I left a comment in the line validation.spec.js:502, The error it's raised exactly here

 ```const invoice = parsePaymentRequest({ request: checkedPrefixlnInvoice });```

But I think it's expected

Reply:
Maybe it could be handled in other way as well

Reply:
yes I tried to fix it like this but it doesn't work
```javascript
it('should return false if the invoice is not valid', async () => {
    let invoice;
    try {
        invoice = await validateInvoice(ctx, 'invalid-invoice');
    } catch (error) {
        invoice = false;
    }
    expect(invoice).to.equal(false);
});
```

Reply:
On that case the expect need to be within the catch

Reply:
Hi @grunch , the test is fine, the message is from the logger that is in the catch of the implementation in the file validations.ts, the error is handled with a return false, that's is why this test passes well.

```
} catch (error) {
    logger.error(error);
    logger.debug(lnInvoice);
    return false;
  }
```
This error could be handled in another way, but regards I mentioned before for the moment is handled with the return false in the catch

Reply:
Hi @grunch , the test is fine, the message is from the logger that is in the catch of the implementation in the file validations.ts, the error is handled with a return false, that's is why this test passes well.

```
} catch (error) {
    logger.error(error);
    logger.debug(lnInvoice);
    return false;
  }
```
This error could be handled in another way, but regards I mentioned before for the moment is handled with the return false in the catch

Reply:
forget about those console.debug lines, I found it on the same function, we can remove it in another PR